### PR TITLE
spec: add extraction error run facet

### PR DESCRIPTION
**Pull request by mobuchowski**
This PR adds `ExtractionErrorRunFacet` which is supposed to reflect internal processing errors of OpenLineage. For example, we want to distinguish SQL job that we parsed and found no datasets processed, from the one we've failed to process.

The design allows describing partial failure. This feature is mainly to use with SQL parser, since it can process multiple statements, yet still do it's job - for example, `ALTER SESSION` statements are not fully supported, yet they don't change the material outcome - we still have proper lineage. 

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

### CODE REVIEW

File: spec/facets/ExtractionErrorRunFacet.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
+	"$schema": "https://json-schema.org/draft/2020-12/schema",
```
Comment (JDarDagran) [inline spec/facets/ExtractionErrorRunFacet.json:2]: Looking at other JSON spec files use 2-space indents.

Comment (JDarDagran) [reply spec/facets/ExtractionErrorRunFacet.json:2]: It requires reformatting anyway I think.

File: spec/facets/ExtractionErrorRunFacet.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    "errors": {
+                        "type": "array",
+                        "items": {
+                            "type" : "object",
+                            "properties" : {
+                                "errorMessage" : {
+                                    "description" : "Text representation of extraction error message.",
+                                    "type" : "string"
+                                },
+                                "task": {
```
Comment (JDarDagran) [inline spec/facets/ExtractionErrorRunFacet.json:28]: How would that look exactly? Let's say parsing SQL fails - should this be a manually-named name of the stage, e.g. `SQL Parser`?

File: spec/facets/ExtractionErrorRunFacet.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                                    "description": "Text representation of task that failed. This can be, for example, SQL statement that parser could not interpret.",
+                                    "type" : "string"
+                                },
+                                "taskNumber": {
+                                    "description": "Order of task (counted from 0).",
```
Comment (JDarDagran) [inline spec/facets/ExtractionErrorRunFacet.json:33]: Is this count of failed tasks or all? If latter, I don't think we have a concept of task in extractors to have them counted in any manner.

**Review by JDarDagran — COMMENTED** [review]
@mobuchowski , it would be good to add new facet to Python client as well.

I also wonder if we could simply put traceback to point with what kind of error we deal.

File: spec/facets/ExtractionErrorRunFacet.json (hunk 3 above)
Comment (mobuchowski) [reply spec/facets/ExtractionErrorRunFacet.json:33]: In this case, this does not relate to Airflow tasks or extractors at all, but would be used as a count of "tasks" inside a single run - piece of work.

A good example is one SQL job that executes multiple statements, whether inside of outside of transaction.

File: spec/facets/ExtractionErrorRunFacet.json (hunk 2 above)
Comment (mobuchowski) [reply spec/facets/ExtractionErrorRunFacet.json:28]: No, this should be a SQL statement that failed.

File: spec/facets/ExtractionErrorRunFacet.json (hunk 1 above)
Comment (mobuchowski) [reply spec/facets/ExtractionErrorRunFacet.json]: Fixed.

**Review by JDarDagran — APPROVED** [review]

**Pull request merged**